tests/networkpolicy_test.go

```diff
@@ -1,173 +1,264 @@
 package tests_test
 
 import (
+	"fmt"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	v1network "k8s.io/api/networking/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
-	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/pkg/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	networkv1 "k8s.io/api/networking/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	netutils "k8s.io/utils/net"
 
 	v1 "kubevirt.io/client-go/api/v1"
 	"kubevirt.io/client-go/kubecli"
 	"kubevirt.io/kubevirt/tests"
 	cd "kubevirt.io/kubevirt/tests/containerdisk"
+	"kubevirt.io/kubevirt/tests/libvmi"
 )
 
 var _ = Describe("[rfe_id:150][crit:high][vendor:cnv-qe@redhat.com][level:component]Networkpolicy", func() {
 
-	var virtClient kubecli.KubevirtClient
-
-	var vmia *v1.VirtualMachineInstance
-	var vmib *v1.VirtualMachineInstance
-	var vmic *v1.VirtualMachineInstance
+	var (
+		virtClient                                          kubecli.KubevirtClient
+		serverVMI, clientVMI, clientVMIAlternativeNamespace *v1.VirtualMachineInstance
+	)
+	Context("when three cirros VMs with default networking are started and serverVMI start an HTTP server on port 80 and 81", func() {
+		tests.BeforeAll(func() {
+			var err error
 
-	tests.BeforeAll(func() {
-		var err error
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
 
-		virtClient, err = kubecli.GetKubevirtClient()
-		tests.PanicOnError(err)
+			tests.SkipIfUseFlannel(virtClient)
+			skipNetworkPolicyRunningOnKindInfra()
+			tests.BeforeTestCleanup()
 
-		tests.SkipIfUseFlannel(virtClient)
-		skipNetworkPolicyRunningOnKindInfra()
-		tests.BeforeTestCleanup()
+			// Create three vmis, serverVMI and clientVMI are in same namespace, clientVMIAlternativeNamespace is in different namespace
+			serverVMI = createVMICirros(virtClient, tests.NamespaceTestDefault, map[string]string{"type": "test"}, libvmi.WithInterface(
```

Comment:
Maybe the VMIs creation should be in a beforeEach (and a delete in an afterEach). I know it will make it much slower, but as far as I know, It is a better practice, and keeps the test cases isolated. 
Do you think it is necessary?

Reply:
i dont think it should be on beforeEach,
because its actually used all over the tests, each test just create a different policy and play with it

Reply:
> i dont think it should be on beforeEach,
> because its actually used all over the tests, each test just create a different policy and play with it

I understand. 
Its isolation vs efficiency and speed. I'm not sure what should we prefer

Reply:
for sure speed and less code, no reason to create destroy imo

Reply:
Since we are not mutating the VMIs at test, we just populate networkpoclies and delete them, we can start them twice, if you we put thins in the limit we can even re-deploy kubevirt at every test and for sure tests will be isolated.